### DISCUSSION TITLE
How to make DateTime parsing consistent?

### DISCUSSION THREAD

**Discussion by EmanuelPoremba** [VerifyTests/Verify#786] [Q&A] [answered]
Hi,
in my situation I have got a an object that contains several properties including DateTime properties. When I disable scrubbing, then those DateTime properties are parsed differently on different machines. So far I failed to set up a custom converter for this (simple?) task.

Situation:

```
[Fact]
    public Task Some_Test()
    {
            return Verify(new
            {
                SomeNumber = 1,
                SomeString = "foo",
                SomeDateTime = new DateTime(2020,1,1)
            });
    }
```

When I run this on my machine I get a datetime formatted by my local machine (01.01.2020), but on our build agent I get another (en-us) datetime format (2020/01/01).
So I tried to add VerifierSettings.TreatAsString<DateTime>((target, _) => ....), but this of course didn't work.

Some other sources stated, that I can set 
> VerifierSettings.AddExtraSettings(_ => _.DateTimeZoneHandling = foo);

but I don't have a property DateTimeZoneHandling to begin with. Furthermore I don't think, that this would solve my formatting issue.

Is there ANY possibility to tell the converter to be culture invariant?

**Comment by SimonCropp** [comment, marked as answer]
i cant reproduce this. can you submit a failing test as a PullRequest.

Note that the date format Very is hard coded to is `yyyy-MM-dd`

eg

```
{
  Date: 2020-10-10 Utc
}
```

https://github.com/VerifyTests/Verify/blob/main/src/Verify.Tests/Serialization/SerializationTests.DontScrubDateTimes.verified.txt

so i need a repro to understand how you are getting `01.01.2020` or `2020/01/01`

**Reply by EmanuelPoremba** [reply]
Hi Simon,

thank you for the rubber duck debugging here. Totally my mistake.
When I tried to setup a standalone solution to share it here I found the issue. I had some .ToString() in the code. That was a fragment from before I learned that there is also a VerifierSettings.DontScrubDateTimes();

Sorry for the stupid question.

**Reply by SimonCropp** [reply]
glad u got it sorted. hope u didnt burn too much time on it
